### DISCUSSION TITLE
Thinking about using this library in HMPO project

### DISCUSSION THREAD

**Discussion by asm2545w** [govuk-react/govuk-react#1144] [Q&A]
Dear all,

We have taken a look into this library and it looks promising for our purpose. One thing we have tried without success was using the GovUK.Radio component to show a nested component when user clicks on it, like the "Conditionally revealing a related question" in GDS [Radios](https://design-system.service.gov.uk/components/radios/). Are you planning to add this functionality? Or maybe it is already in place but we could not find it.

Another thing we could not handle was to make GovUK.Radio or GovUK.Checkbox smaller, like using 'govuk-radios--small' GDS class. Are we missing something?

Regards,
Alvaro

**Comment by penx** [comment]
Thanks for getting in touch!

> We have taken a look into this library and it looks promising for our purpose. One thing we have tried without success was using the GovUK.Radio component to show a nested component when user clicks on it, like the "Conditionally revealing a related question" in GDS [Radios](https://design-system.service.gov.uk/components/radios/). Are you planning to add this functionality? Or maybe it is already in place but we could not find it.

I don't think our Radio component allows for this layout at the moment but I'd consider merging a PR if you'd like to contribute something that matches govuk-frontend.

The code in govuk-react would not handle the state for this use case but would provide the styles, components and props needed for the consuming application to achieve this functionality.

> Another thing we could not handle was to make GovUK.Radio or GovUK.Checkbox smaller, like using 'govuk-radios--small' GDS class. Are we missing something?

This is covered by open ticket https://github.com/govuk-react/govuk-react/issues/597 which is currently scheduled for v2.1.0 (so some time off) but if you want to contribute a PR for this in the meantime I could consider merging it in sooner.

**Comment by asm2545w** [comment]
Dear,

We have finally decided to use our components for now, but we could
consider using govuk-react in the future. Thanks for getting back to us.

Regards,
Alvaro San Millan
